tests/test_virtual_chassis.py

```diff
@@ -136,7 +136,6 @@ def test_voq_switch(self, vct):
                 spcfg = ast.literal_eval(value)
                 assert spcfg['count'] == sp_count, "Number of systems ports configured is invalid"
 
-    @pytest.mark.skip(reason="This test is not stable enough")
```

Comment:
Can I have the context for this? 

`@pytest.mark.skip(reason="This test is not stable enough")`

Why was it added? Was it added after https://github.com/Azure/sonic-buildimage/pull/7629 merged? Did https://github.com/Azure/sonic-buildimage/pull/7629 cause test instability?

Reply:
OK. I saw you mentioned 2 reasons.

(1) The changes from PR Azure/sonic-buildimage#7857 introduced problem in construction of config_db.json from /usr/share/sonic/virtual_chassis/default_config.json that is needed for VS virtual chassls tests.

(2) The fabricportlane mapping introduced by PR Azure/sonic-buildimage#7629. Along with this PR merging FABRIC_NUM_PORTS in conftest.py should be set to 16 (as stated in PR #1459)

I didn't look at carefully reason 1, but I'm curious why reason 2 could cause test failure?

It's because I think that if FABRIC_NUM_PORTS = 0, it still works fine. Of course, we have to change to 16 to test fabric code, but the value of 0 should not cause test failures.

It's because we are testing at least n ports, not exact n ports. Please find the following code in `tests/conftest.py`.

```
        # Verify that we have "at least" NUM_PORTS + FABRIC_NUM_PORTS, rather exact number.
        # Right now, FABRIC_NUM_PORTS = 0. So it essentially waits for at least NUM_PORTS.
        # This will allow us to merge Azure/sonic-buildimage#6185 that creates 16 fabric ports.
        # When PR#6185 merges, FABRIC_NUM_PORTS should be 16, and so this verification (at least
        # NUM_PORTS) still holds.
        # Will update FABRIC_NUM_PORTS to 16, and revert back to wait exact NUM_PORTS + FABRIC_NUM_PORTS
        # when PR#6185 merges.

        wait_at_least_n_keys = True
        asic_db.wait_for_n_keys("ASIC_STATE:SAI_OBJECT_TYPE_PORT", num_ports + 1, wait_at_least_n_keys)  # +1 CPU Port

        # Verify that fabric ports are monitored in STATE_DB
        if metadata.get('switch_type', 'npu') in ['voq', 'fabric']:
            self.get_state_db()
            self.state_db.wait_for_n_keys("FABRIC_PORT_TABLE", FABRIC_NUM_PORTS, wait_at_least_n_keys)
```

`wait_at_least_n_keys` means we are verifying at least `NUM_PORTS + FABRIC_NUM_PORTS`, not exact `NUM_PORTS + FABRIC_NUM_PORTS`. Since FABRIC_NUM_PORTS = 0, the verification still returns true when Azure/sonic-buildimage#7629 merged.

And could you please update the above code in `tests/conftest.py` by removing the comment and `wait_at_least_n_keys` in the call `wait_for_n_keys`? I think we are now safe to verify the exact number of ports.

Reply:
Those 3 tests consistently failed. So PR https://github.com/Azure/sonic-swss/pull/1807 was raised to skip these tests.
These tests failed not because of  Azure/sonic-buildimage#7629 but because of https://github.com/Azure/sonic-buildimage/pull/7857. I noticed that once the problem created by https://github.com/Azure/sonic-buildimage/pull/7857 is fixed, the port count mismatch problem occurred, which was due to Azure/sonic-buildimage#7629

Reply:
Cool, thanks.